### PR TITLE
[ENH] Adapt the DCNN Networks to use Weight Norm Wrappers

### LINKED ISSUE #1714: [ENH] Add Weight Normalization layer using Tensorflow
### Describe the feature or idea you want to propose

Implement Weigh Normalization layer using the tensorflow core API. To be used later in the DCNN Network. 

References: [https://www.tensorflow.org/addons/api_docs/python/tfa/layers/WeightNormalization](https://www.tensorflow.org/addons/api_docs/python/tfa/layers/WeightNormalization)

### PR DESCRIPTION
**Pull request by aadya940**
Apply weight norm wrappers and remove L2 Regularization. 

Closes #1714

### CODE REVIEW

**Comment by aadya940** [comment]
@hadifawaz1999 LGTM!

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Seems you forgot to also do it for the decoder dcnn layers in AEDCNNNetwork, other than that looks great

**Comment by aadya940** [comment]
Hey @hadifawaz1999 , This layer doesn't work well with `Conv1DTranspose`, I think thats because our `_WeightNormalization` wrapper internally uses `tf.nn.conv1d`.

**Comment by hadifawaz1999** [comment]
oh okay i see i see, [originally](https://github.com/blafabregue/TimeSeriesDeepClustering/blob/main/networks/dilated_causal_cnn.py) conv transposes were not yet a thing so they didnt have an issue. Lets get this in then and we'll think of a solution for later.

**Review by hadifawaz1999 — APPROVED** [review]
